manta-crypto/src/eclair/mod.rs

```diff
@@ -46,6 +46,22 @@ impl Native for () {
 /// This `trait` is explicitly not implemented for `()`, the default native compiler. This marker
 /// can be used to write explicitly different implementations for native and non-native compilers
 /// where otherwise a generic implementation would have to exist.
+///
+/// # Limitations
+///
+/// This is an emulation of an unimplemented feature of rust called ["negative trait bounds"]. As it
+/// currently stands, the compiler will make instances of [`NonNative`] outside of the ECLAIR crate
+/// unusable as you'll run into [error 0119] which has the following notice:
```

Comment:
The main issue here is that `NonNative` (and its implementations for the native Rust types like `()`) is controlled by the `manta_crypto` crate. Adding an implementation for `NonNative` to `()` is not considered a breaking change by the Rust compiler since we're strictly increasing the number of interfaces that can interact with each other, not changing or removing existing ones. However, for a downstream crate that would use `NonNative` to give different implementations for `()` and all other `COM` arguments, this _is_ in fact a breaking change. In order to preserve the API breakage rules, Rust just does not allow the downstream crate to use `NonNative` as ECLAIR uses it.